cosmwasm/README.md

```diff
@@ -1,45 +1,93 @@
 
+## Deployment
+
+Deploying the cosmwasm contract has two steps:
+1. Upload the code. This step will give you a code id.
+2. Either create a new contract, or migrate an existing one:
+    1. Creating a new contract which has an address with a code id as its program.
+    2. Migrating an existing contract code id to the new code id.
+
+This directory includes a script to perform both steps. Read below for the details.
+
+### Uploading the code
+
+First build the contracts within [the current directory](./):
 
 ``` sh
-docker build -f Dockerfile.build -o artifacts .
+bash build.sh
 ```
 
-Then, for example, to deploy `token_bridge.wasm`, run in the `tools` directory
+This command will build and save Pyth contract in the `artifact` directory.
+
+Then, to deploy Pyth contract (`pyth_bridge.wasm`), run the following command in the `tools` directory:
 
 ``` sh
-npm ci
-node deploy_single.js --network mainnet --artifact ../artifacts/token_bridge.wasm --mnemonic "..."
+npm ci # Do it only once to install required packages
+npm run deploy-pyth -- --network testnet --artifact ../artifacts/pyth_bridge.wasm --mnemonic "..."
 ```
 
-which will print something along the lines of
+If successful, this command will print something along the lines of:
 
 ``` sh
-Storing WASM: ../artifacts/token_bridge.wasm (367689 bytes)
+Storing WASM: ../artifacts/pyth_bridge.wasm (367689 bytes)
 Deploy fee:  88446uluna
 Code ID:  2435
 ```
 
-# Migrate
+If you do not pass any additional arguments to the script, it will only upload the code and return the code id. If you want to create a 
+new contract, or upgrade an existing contract you should pass more arguments that are described below.
 
-## Mainnet
+### Instantiating new contract
+If you want instantiate a new contract after your deployment, pass `--instantiate` argument to the above command.
+This command will upload the code and with the resulting code id instantiates a new pyth contract:
 
-Migrations on mainnet have to go through governance. Once the guardians sign the
-upgrade VAA, the contract can be upgraded by submitting the signed VAA to the
-appropriate contract. For example, to upgrade the token bridge on mainnet,
-in `wormhole/clients/token_bridge/`:
+``` sh
+npm run deploy-pyth -- --network testnet --artifact ../artifacts/pyth_bridge.wasm --mnemonic "..." --instantiate
+```
+
+If successful, the output should look like:
+```
+Storing WASM: ../artifacts/pyth_bridge.wasm (183749 bytes)
+Deploy fee:  44682uluna
+Code ID:  53199
+Instantiating a contract
+Sleeping for 10 seconds for store transaction to finalize.
+Instantiated Pyth Bridge at terra123456789yelw23uh22nadqlyjvtl7s5527er97 (0x0000000000000000000000001234567896267ee5479752a7d683e49317ff4294)
+Deployed pyth contract at terra123456789yelw23uh22nadqlyjvtl7s5527er97
+```
+
+### Migrating existing contract
+If you want to upgrade an existing contract pass `--migrate --contract terra123456xyzqwe..` arguments to the above command.
```

Comment:
i think this either needs to say "pass the `...` arguments to the" or just "pass `...` to the". I think the second is better

Reply:
yeah I agree; the second one is better.